### PR TITLE
feat(engine): Add configuration for filter sent macros

### PR DESCRIPTION
**Pull request by leoncx**
<h1> Pull Request Template </h1>

<h2> Description </h2>

* Better display of Broker Options on Centreon Engine configuration
* Add filter for sent macros from Centreon Engine to Centreon Broker

Refs: MON-3629

<h2> Type of change </h2>

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

<h2> Target serie </h2>

- [ ] 2.8.x
- [ ] 18.10.x
- [X] 19.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

<h2> Checklist </h2>

<h5> Community contributors & Centreon team </h5>

- [X] I followed the **coding style guidelines** provided by Centreon
- [X] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [X] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

<h5> Centreon team only </h5>

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests covers 80%** of the code written for the story.
- [X] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/configuration/configNagios/formNagios.php [hunk 1]
```diff
@@ -758,6 +850,28 @@ function isNum($value)
 $form->registerRule('exist', 'callback', 'testExistence');
 $form->registerRule('isNum', 'callback', 'isNum');
 
+/* Add validator for macro name format */
+/**
+ * Valide the macro name
```
Comment (sc979) [inline www/include/configuration/configNagios/formNagios.php]: Valide -> Validate

File: www/class/centreon-config/centreonMainCfg.class.php [hunk 2]
```diff
@@ -42,6 +42,39 @@ class CentreonMainCfg
 
     private $DB;
 
+    // List of broker options Centreon Engine
+    // (https://documentation.centreon.com/docs/centreon-engine/en/latest/user/configuration/basics/main_configuration_file_options.html#event-broker-options)
```
Comment (loiclau) [inline www/class/centreon-config/centreonMainCfg.class.php]: I thinks that line is too long

File: www/class/config-generate/engine.class.php [hunk 3]
```diff
@@ -266,7 +270,7 @@ class Engine extends AbstractObject
         'enable_predictive_service_dependency_checks',
         'use_large_installation_tweaks',
         'enable_environment_macros',
-        'use_setpgid', # cengine
+        'use_setpgid', # centengine
```
Comment (loiclau) [inline www/class/config-generate/engine.class.php]:
```suggestion
        'use_setpgid', // centengine
```

File: www/class/config-generate/engine.class.php [hunk 4]
```diff
@@ -220,7 +222,7 @@ class Engine extends AbstractObject
         //'debug_level_opt', ????!!
```
Comment (loiclau) [inline www/class/config-generate/engine.class.php]:
```suggestion
```

File: www/include/configuration/configNagios/formNagios.php [hunk 5]
```diff
@@ -766,22 +882,14 @@ function isNum($value)
 $form->addRule('cfg_file', _("Required Field"), 'required');
 $form->addRule('nagios_comment', _("Required Field"), 'required');
 $form->addRule('nagios_name', _("Name is already in use"), 'exist');
+/* Add rule to field for whitelist macro */
```
Comment (loiclau) [inline www/include/configuration/configNagios/formNagios.php]:
```suggestion
// Add rule to field for whitelist macro
```

**Pull request merged**